spark3/src/main/java/org/apache/iceberg/spark/Spark3Util.java

```diff
@@ -579,6 +579,8 @@ public String or(String leftResult, String rightResult) {
           return pred.ref().name() + " != " + sqlString(pred.literal());
         case STARTS_WITH:
           return pred.ref().name() + " LIKE '" + pred.literal() + "%'";
+        case NOT_STARTS_WITH:
+          return pred.ref().name() + " NOT LIKE '" + pred.literal() + "%'";
```

Comment:
Here's another case where we possibly wind up with too many quotes. I can follow up in another PR. I chose to go with the existing format.

Reply:
Resolving this coversation as there is already a follow up issue https://github.com/apache/iceberg/issues/2059. I'll mention that `Spark3Util` shouuld be considered when working on that issue as a follow up.